table/block_based/block_based_table_factory.cc

```diff
@@ -419,6 +423,12 @@ BlockBasedTableFactory::BlockBasedTableFactory(
     : table_options_(_table_options) {
   InitializeOptions();
   RegisterOptions(&table_options_, &block_based_table_type_info);
+
+  if (table_options_.reserve_table_reader_memory &&
+      table_options_.no_block_cache == false) {
```

Comment:
[Note] A fun little thing I found is when `table_options_.no_block_cache` is set to 1 by option string, the negate of it `!table_options_.no_block_cache` in dbg build (but not DEBUG_LEVEL=1) evaluates to `True` incorrectly ......